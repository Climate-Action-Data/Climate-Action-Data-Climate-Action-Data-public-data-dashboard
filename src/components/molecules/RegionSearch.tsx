import { Regions } from "@/@types/Region"
import { Flex, Text, Box, Divider, HStack, Tag, Button, TagLabel } from "@chakra-ui/react"
import { useState } from "react"

export const RegionSearch = (): React.JSX.Element => {
    const [region, setRegion] = useState<string | undefined>(undefined)
    return (
        <Flex height={"40px"} alignItems={"center"}>
            {region
                ? <Flex alignItems={"center"}><Button>&lt;</Button><Text>{region}</Text></Flex>
                : <Box><Text fontWeight={"bold"}>Region</Text></Box>
            }
            <Divider marginX={6} height={"70%"} orientation="vertical" />
            <HStack spacing={4}>
                {Regions.map((region) => (
                    <Button variant={"unstyled"} onClick={() => setRegion(region)}>

                        <Tag
                            size={"md"}
                            key={region}
                            borderRadius='full'
                            variant='solid'
                            colorScheme='gray'
                        >
                            <TagLabel>{region}</TagLabel>
                        </Tag>
                    </Button>

                ))}
            </HStack>
        </Flex>
    )
}