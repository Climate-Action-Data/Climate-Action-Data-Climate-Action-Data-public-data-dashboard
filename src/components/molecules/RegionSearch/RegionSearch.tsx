import { Regions } from "@/@types/Region"
import { Flex, Text, Box, Divider, HStack, Tag, Button, TagLabel } from "@chakra-ui/react"
import { useState } from "react"


export const RegionSearch = (): React.JSX.Element => {
    const [region, setRegion] = useState<string | undefined>(undefined)
    const regionTitle = () => {
        if (region) {
            return (<Flex alignItems={"center"}>
                <Button borderRadius={"50%"} backgroundColor={"transparent"} onClick={() => setRegion(undefined)}>&lt;</Button>
                <Text fontWeight={"bold"}>{region}</Text>
            </Flex>)
        } else {
            return <Box><Text fontWeight={"bold"}>Region</Text></Box>
        }
    }
    return (
        <Flex height={"40px"} alignItems={"center"}>
            {regionTitle()}
            <Divider marginX={6} height={"70%"} orientation="vertical" />
            <HStack spacing={4}>
                {Regions.map((regionName) => (regionName !== region &&
                    <Button key={regionName} variant={"unstyled"} onClick={() => setRegion(regionName)}>
                        <Tag
                            size={"md"}
                            borderRadius='full'
                            variant='solid'
                            colorScheme='gray'
                        >
                            <TagLabel>{regionName}</TagLabel>
                        </Tag>
                    </Button>

                ))}
            </HStack>
        </Flex>
    )
}