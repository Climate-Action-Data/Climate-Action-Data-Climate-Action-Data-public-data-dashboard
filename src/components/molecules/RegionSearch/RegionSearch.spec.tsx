import { render } from '@testing-library/react'
import { RegionSearch } from './RegionSearch';

it('renders correctly', () => {
    const { container } = render(<RegionSearch />)
    expect(container).toMatchSnapshot();
});