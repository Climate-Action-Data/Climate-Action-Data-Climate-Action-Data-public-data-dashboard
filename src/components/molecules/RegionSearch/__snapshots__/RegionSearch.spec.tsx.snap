// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-hajjsy"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-in3yi3"
      >
        Region
      </p>
    </div>
    <hr
      aria-orientation="vertical"
      class="chakra-divider css-8w54o2"
    />
    <div
      class="chakra-stack css-1mlvmbj"
    >
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        <span
          class="css-1ehzxp"
        >
          <span
            class="css-zvlevn"
          >
            North America
          </span>
        </span>
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        <span
          class="css-1ehzxp"
        >
          <span
            class="css-zvlevn"
          >
            South America
          </span>
        </span>
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        <span
          class="css-1ehzxp"
        >
          <span
            class="css-zvlevn"
          >
            Europe
          </span>
        </span>
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        <span
          class="css-1ehzxp"
        >
          <span
            class="css-zvlevn"
          >
            Africa
          </span>
        </span>
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        <span
          class="css-1ehzxp"
        >
          <span
            class="css-zvlevn"
          >
            Asia
          </span>
        </span>
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        <span
          class="css-1ehzxp"
        >
          <span
            class="css-zvlevn"
          >
            ANZ
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
